pkg/compare/capturegroupsInlineDiff_test.go

```diff
@@ -310,3 +399,40 @@ func TestCapturegroupsDiff(t *testing.T) {
 		})
 	}
 }
+
+func TestCapturegroupsValidate(t *testing.T) {
+	suites := []struct {
+		pattern     string
+		expectError bool
+	}{
+		{
+			pattern:     "",
+			expectError: false,
+		},
+		{
+			pattern:     "No regex",
+			expectError: false,
+		},
+		{
+			pattern:     "No capturegroups (.*) but some regex",
+			expectError: false,
+		},
+		{
+			pattern:     "Simple capturegroup (?<named>.*)",
+			expectError: false,
+		},
+		{
+			pattern:     "Broken capturegroup (?<named>[[:_broken_character_class_name_:]])",
```

Comment:
It was surprisingly difficult to find something that looked enough like a capture group to be validated but also wrong and fails actual regex compilation.  

If anyone has ideas for additional cases I'd be happy to add them!